src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -830,21 +835,7 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         }
         _interactivity.Initialize();
 
-        _AttachDxgiSwapChainToXaml(reinterpret_cast<HANDLE>(_core.SwapChainHandle()));
-
-        // Tell the DX Engine to notify us when the swap chain changes. We do
-        // this after we initially set the swapchain so as to avoid unnecessary
-        // callbacks (and locking problems)
         _core.SwapChainChanged({ get_weak(), &TermControl::RenderEngineSwapChainChanged });
-
-        // !! LOAD BEARING !!

```

Comment:
don't _love_ the removal of `LOAD BEARING` warning comments...

Reply:
They cause problems though... `EnablePainting()` is not a thread synchronization primitive. Creating the resources on the main thread here and then calling `EnablePainting()` doesn't ensure that the render thread can actually access the most up to date representation of those classes.